pkg/utils/utils.go

```diff
@@ -137,42 +139,36 @@ func RetrieveVcConfigSecret(params map[string]interface{}, config *rest.Config,
 		return errors.New(errMsg)
 	}
 
-	var sEnc string
-	var lines []string
+	ParseConfig(secret, params, logger)
+
+	return nil
+}
+
+func ParseConfig(secret *k8sv1.Secret, params map[string]interface{}, logger logrus.FieldLogger) {
+	// Setup config
+	var conf vcConfig.Config
+	// Read secret data
 	for _, value := range secret.Data {
-		sEnc = string(value)
-		lines = strings.Split(sEnc, "\n")
-		logger.Debugf("Successfully retrieved vCenter configuration from secret %s", secret.Name)
+		confStr := string(value) // Convert the secret data to a string
+		err := gcfg.FatalOnly(gcfg.ReadStringInto(&conf, confStr))
+		if err != nil {
+			logger.WithError(err).Error("Failed to parse vSphere secret data")
+		}
+		logger.Debugf("Successfully parsed vCenter configuration from secret %s", secret.Name)
 		break
 	}
 
-	ParseLines(lines, params, logger)
+	// Use config data from struct to populate params map passed by RetrieveVcConfigSecret callers
+	params["cluster-id"] = conf.Global.ClusterID
```

Comment:
If there was an error(the one you ignored), conf wouldn't be populated, this could lead to nil pointer errors.

Reply:
If conf is not populated, then VC connection would fail. The logs would show that the failure is resulting from a failure to parse the vSphere config secret.